docs/demo.md

```diff
@@ -0,0 +1,218 @@
+</Canvas>
+
+## Controlled
+
+<Canvas>
+  <Story name="Controlled">
+    {args => {
+      const updateArgs = useArgs()[1];
+      const handleEvent = argName => updateArgs({ argName });
+      return <ComponentStory {...args} onEvent={handleEvent} />;
+    }}
+  </Story>
+</Canvas>
+```
+
+Note how the `ComponentStory` is reused for both canvas stories. See
+[Story](#story) below for additional details. Be sure to move common `args`,
+`argTypes`, and `parameters` under the MDX `<Meta>` rather than repeating these
+under each story.
+
+#### Story `args`
+
+- Storybook automatically provides main component props as `args`. Usually there
+  is no need to specify these values (double-check for missing prop controls). One
+  notable exception is for Garden boolean prop values that default to `true`. In
+  this case, the `args` should be specified to match.
+- Select smart defaults for the remaining subcomponent or "Story" `args`.
+  Sometimes the best choice is to leave an arg undefined.
+
+#### Story `argTypes`
+
+Standard convention for `argTypes` exists across dozens of existing stories.
+Study demo code for details. Most importantly, main component props that are
+_not_ auto-generated must be properly categorized. The two acceptable story
+categories are:
+
+- Subcomponent. All subcomponent props (including `children`) are specified
+  under the associated subcomponent name.
+- "Story". This category includes all ancillary controls that are needed to
+  stress the full flexibility of the rendered component(s).
+
+#### Story `parameters`
+
+Currently, story `parameters` are used to link to the Figma page(s) or frame(s)
+that best exhibits associated component designs. Note that the parameter URLs
+are internal-only and point to designs that exist in the main Garden Figma
+branch. Therefore, `parameters` may need to be added as a follow-on for new
+components.
+
+### Story
+
+A `Story` type component is the best way to develop demo code for non-trivial
+Garden components. Doing so removes complexity of coding in MDX, placing
+component rendering and application of story args into TypeScript where code can
+be properly maintained. In its basic form, a `Story` looks like this:
+
+```tsx
+import React from 'react';
+import { Story } from '@storybook/react';\
+import { Component, IComponentProps } from '@zendeskgarden/react-package';
+
+interface IArgs extends IComponentProps {
+  /* Subcomponent and "Story" arg definitions go here */
+}
+
+export const ComponentStory: Story<IArgs> = args => (
+  <Component {...args} />
+);
+```
+
+Take time to ensure Story `args` and `argTypes` are defined well. Often you will
+find that the majority of demo development time is spent on `args` and
+`argTypes` and the `Story` itself essentially falls out of the goal to
+demonstrate component flexibility in isolation.
+
+## Naming
+
+- Follow the existing Storybook naming hierarcy:
+  `Packages/{{PackageName}}/{{ComponentName}}`
```

Comment:
What is the idea behind `Packages` instead of `Component`?

Reply:
Per https://github.com/zendeskgarden/react-components/pull/1265#discussion_r778096091, the naming reflects the basic structure of `react-components`. Also, when we find ourselves setting `{{PackageName}}` as the second level, basic taxonomy science should tell us that the root level is "Packages" (plural).

Reply:
If Garden ever decides to deviate the basic package structure, then this naming will need to change. For now, it clearly tells us which components are part of which packages.